src/transport/NetworkProvisioning.cpp

```diff
@@ -187,6 +202,26 @@ CHIP_ERROR NetworkProvisioning::SendIPAddress(const Inet::IPAddress & addr)
     return err;
 }
 
+CHIP_ERROR NetworkProvisioning::SendCurrentIPv4Address()
+{
+    for (chip::Inet::InterfaceAddressIterator it; it.HasCurrent(); it.Next())
```

Comment:
At some point we probably want to propagate the source interface as well.

That seems really important for link-local and I imagine there could be unusual network setups where different network cards exist on different LANs.

Probably a TODO for the future.

Reply:
@andy31415, this code `NetworkProvisioning.h/.cpp` will be deleted once we move the provisioning to ZCL cluster and DNS-SD. I'll file a GH issue in regards to that.

Reply:
Filed #4606